### ISSUE TITLE
Build the BVAR model for inflation and Growth

### ISSUE THREAD

**Issue by devminda** [MSFE26-FIN-589/APM#10]

**Comment by devminda** [comment]
<img width="1189" height="490" alt="Image" src="https://github.com/user-attachments/assets/ca1e652f-1c8a-44c6-ba9b-aa8531ae5c3f" />

The test accuracy in terms of RMSE:  5.583 on the test set.
Beat the Naive Benchmark by 40%


p05 | p16 | median | p84 | p95
-- | -- | -- | -- | --
49.017252 | 51.111965 | 54.042974 | 57.007434 | 59.054380
47.824955 | 50.366858 | 54.067705 | 57.841360 | 60.392297
46.301957 | 49.579415 | 53.911609 | 58.360987 | 61.067484
45.143886 | 48.624596 | 53.729661 | 58.665073 | 62.069142
44.254588 | 47.738170 | 53.587912 | 58.803639 | 62.609929
42.925070 | 47.257201 | 53.436121 | 59.010513 | 62.948426

**Comment by devminda** [comment]
<details>
<summary><strong> Full Data Engineering & Modeling Pipeline</strong></summary>

<br>

1. **Start with messy raw data (`Growth_raw.xlsx`)**
   - Each variable (ISM, T10Y2Y, SPX, jobless claims, NFCI, permits, copper) lived in its own sheet.
   - Mixed frequencies:
     - Daily → SPX, T10Y2Y
     - Weekly → jobless claims, NFCI
     - Monthly → ISM, permits, copper
   - Some sheets contained corrupted entries (e.g., copper had literal `"####"` strings saved as text).

2. **Clean corrupted and inconsistent data**
   - Convert `"####"` and other non‑numeric artifacts into proper missing values.
   - Interpolate *only small gaps* where values on both sides exist.
   - Never fabricate long stretches of missing data.
   - Ensure all numeric columns are parsed as floats.

3. **Align everything to a monthly calendar**
   - ISM is monthly → all other series must be converted to monthly.
   - For daily/weekly series, compute:
     - `_AVG` → monthly average (captures “how the month behaved overall”)
     - `_EOM` → end‑of‑month snapshot (captures “the final state of the month”)
   - Monthly series (permits, copper) simply aligned to month‑end.
   - Final calendar begins in **September 1996**, the first ISM observation.

4. **Choose one version per variable for the BVAR (avoid multicollinearity)**
   - `_AVG` and `_EOM` versions of the same variable are almost perfectly correlated.
   - Including both would add parameters without adding information.
   - Final choices:
     - **End‑of‑month (EOM)** → snapshot variables  
       - T10Y2Y (yield spread)  
       - SPX (S&P 500)
     - **Monthly average (AVG)** → flow/condition variables  
       - Jobless claims  
       - NFCI
     - **Building permits** → already monthly, no choice needed.

5. **Drop copper**
   - Most recent 12 months were corrupted beyond repair.
   - Historically weak relationship with ISM.
   - Keeping it would shorten the usable dataset for no meaningful gain.

6. **Handle the COVID shock properly**
   - Early attempts manually smoothed 2020 spikes.
   - Final approach: keep the *real* data but tell the BVAR which months were abnormal.
   - Using the library:
     - `covid=True`
     - `covid_dates=['2020-02','2020-12']`
   - This creates **one dummy per COVID month** (11 total).
   - Each dummy = 1 only in its specific month, 0 elsewhere.
   - This gives the model a separate adjustment knob for each pandemic month.
   - Diagnostics show large dummy coefficients → model is actively absorbing COVID noise.

7. **Transform variables for modeling**
   - **Keep in levels** (bounded, mean‑reverting):
     - ISM  
     - T10Y2Y  
     - NFCI  
   - **Log‑transform** (long‑run upward trend):
     - Jobless claims  
     - SPX  
     - Building permits  
   - This stabilizes variance and improves VAR behavior.

8. **Combine everything into one unified monthly dataset**
   - All variables merged into a single DataFrame.
   - Index = month‑end dates.
   - No missing values allowed in the final modeling set.

9. **Split into training and test sets**
   - Hold out the most recent **24 months**.
   - Model never sees the test period during estimation.
   - Allows honest forecast evaluation instead of overfitting.

<br>

</details>

**Comment by devminda** [comment]
<img width="1189" height="490" alt="Image" src="https://github.com/user-attachments/assets/49147e33-2886-4131-82df-e5d8d7438c20" />

Best naive benchmark RMSE: 22.456
BVAR does NOT beat the best naive benchmark on this test window.

## Forecast


| date        | p05        | p16        | median     | p84        | p95        |
|-------------|------------|------------|------------|------------|------------|
| 2026-09-01  | 266.312954 | 271.995056 | 280.977665 | 290.095877 | 296.250784 |
| 2026-10-01  | 258.806580 | 267.927415 | 282.779525 | 297.576676 | 307.980997 |
| 2026-11-01  | 251.844505 | 264.336996 | 283.119341 | 304.278051 | 318.236905 |
| 2026-12-01  | 247.229118 | 260.694968 | 283.873173 | 308.775290 | 327.197371 |
| 2027-01-01  | 242.778309 | 256.641419 | 284.743155 | 312.987660 | 334.201815 |
| 2027-02-01  | 236.329747 | 254.652010 | 284.531964 | 318.083270 | 341.991375 |



**I would not recommend this for inflation. It did not perform well on the data**
